app/models/server.py

```diff
@@ -13,10 +13,9 @@ class Server(APIDocument):
 
     owner = fields.ReferenceField(User, required=True)
     join_rules = fields.ListField(fields.ReferenceField("ServerJoinRule"), default=[], required=False)
-
+    server_token = fields.StrField(required=False)
```

Comment:
what's the thinking behind these `server_token` and `owns_token` fields?

Reply:
Gotcha. In that case, we already have some building blocks done on this and it'd probably be best to take these changes into a diff discussion (and PR) altogether.

FYI, if you look at the method `is_eligible_to_join_server()` you'll find some logic for delegating token validation to [guild.xyz](https://www.guild.xyz/). Storing and handling all that info ourselves will be quite complex I believe.